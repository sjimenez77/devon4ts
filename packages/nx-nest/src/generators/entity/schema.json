{
  "$schema": "https://json-schema.org/schema",
  "$id": "Entity",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    },
    "projectName": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 1
      },
      "x-prompt": "In which project would you like to add the entity?"
    },
    "module": {
      "type": "string",
      "description": "Module where the entity will be generated"
    }
  },
  "required": ["name", "projectName"]
}
