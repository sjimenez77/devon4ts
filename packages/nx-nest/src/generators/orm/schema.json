{
  "$schema": "http://json-schema.org/schema",
  "$id": "InitTypeormSchematic",
  "title": "Init Typeorm Schema",
  "type": "object",
  "properties": {
    "orm": {
      "type": "string",
      "enum": ["typeorm"],
      "default": "typeorm"
    },
    "db": {
      "type": "string",
      "enum": ["postgres", "cockroachdb", "mariadb", "mysql", "sqlite", "oracle", "mssql", "mongodb"],
      "default": "sqlite"
    },
    "projectName": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "In which project would you like to add typeorm configuration?"
    }
  },
  "required": ["projectName"]
}
