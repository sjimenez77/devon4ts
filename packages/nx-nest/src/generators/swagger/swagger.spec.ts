import { Tree, readProjectConfiguration } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { ApplicationGeneratorOptions } from '@nx/nest/src/generators/application/schema';
import applicationGenerator from '../application/application';
import convictGenerator from '../convict/convict';
import { swaggerGenerator } from './swagger';

describe('swagger generator', () => {
  let tree: Tree;
  const options: ApplicationGeneratorOptions = {
    name: 'test',
    projectNameAndRootFormat: 'as-provided',
    directory: 'apps/test',
  };

  beforeAll(async () => {
    tree = createTreeWithEmptyWorkspace();
    await applicationGenerator(tree, options);
  }, 60000);

  it('should run successfully', async () => {
    const config = readProjectConfiguration(tree, 'test');
    expect(config).toBeDefined();
  });

  describe('common swagger generator tasks', () => {
    beforeAll(async () => {
      await swaggerGenerator(tree, { projectName: options.name });
    }, 60000);

    it('should add dependencies to package.json', async () => {
      const fileContent = tree.read('package.json')?.toString('utf-8');
      expect(fileContent).toMatch(/"dependencies": {(.|\n)*"@nestjs\/swagger":/g);
    });

    it('should add Swagger configuration to main.ts', async () => {
      const fileContent = tree.read(`./packages/nx-nest/apps/${options.name}/src/main.ts`)?.toString('utf-8');
      expect(fileContent).toMatchSnapshot();
    });

    it('should not have BaseEntity model', async () => {
      const filePath = `./packages/schematcis/apps/${options.name}/src/app/shared/model/entities/base.entity.ts`;
      expect(tree.exists(filePath)).toBeFalsy();
    });

    it('should not have config.ts', async () => {
      const filePath = `./packages/schematcis/apps/${options.name}/src/config.ts`;
      expect(tree.exists(filePath)).toBeFalsy();
    });
  });

  describe('swagger generator with convict', () => {
    beforeAll(async () => {
      await convictGenerator(tree, { projectName: options.name });
      await swaggerGenerator(tree, { projectName: options.name });
    }, 60000);

    it('should add Swagger properties to config.ts', async () => {
      const filePath = `./packages/nx-nest/apps/${options.name}/src/config.ts`;
      const fileContent = tree.read(filePath)?.toString('utf-8');
      expect(fileContent).toMatchSnapshot();
    });

    it('should add Swagger types at app-config', async () => {
      const filePath = `./packages/nx-nest/apps/${options.name}/src/app/app-config.ts`;
      const fileContent = tree.read(filePath)?.toString('utf-8');
      expect(fileContent).toMatchSnapshot();
    });
  });
});
