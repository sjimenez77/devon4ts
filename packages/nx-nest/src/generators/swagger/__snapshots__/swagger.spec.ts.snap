// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`swagger generator common swagger generator tasks should add Swagger configuration to main.ts 1`] = `
"/**
 * This is not a production server yet!
 * This is only a minimal backend to get started.
 */

import { Logger, ValidationPipe, VersioningType } from '@nestjs/common';
import { NestFactory } from '@nestjs/core';

import { AppModule } from './app/app.module';
import { WinstonLogger } from '@proj/shared/logger';
import { DocumentBuilder, SwaggerModule } from '@nestjs/swagger';

async function bootstrap(): Promise<void> {
  const app = await NestFactory.create(AppModule, { bufferLogs: true });

  const logger = await app.resolve(WinstonLogger);
  app.useLogger(logger);

  const globalPrefix = 'api';
  app.setGlobalPrefix(globalPrefix);
  const port = process.env.PORT ?? 3000;
  app.useGlobalPipes(
    new ValidationPipe({
      transform: true,
      transformOptions: {
        excludeExtraneousValues: true,
      },
    }),
  );
  app.enableVersioning({
    type: VersioningType.URI,
    defaultVersion: '1',
  });
  if (process.env.NODE_ENV !== 'production') {
    const options = new DocumentBuilder()
      .setTitle('NestJS application')
      .setDescription('')
      .setVersion('0.0.1')
      .addBearerAuth()
      .build();

    const swaggerDoc = SwaggerModule.createDocument(app, options);
    SwaggerModule.setup(globalPrefix + '/v1/openapi', app, swaggerDoc);
    Logger.log(\`üç™ OpenAPI is running on http://localhost:\${port}/\${globalPrefix}/v1/openapi\`);
  }
  await app.listen(port);
  Logger.log(\`üöÄ Application is running on: http://localhost:\${port}/\${globalPrefix}\`);
}

bootstrap();
"
`;

exports[`swagger generator swagger generator with convict should add Swagger properties to config.ts 1`] = `
"import { config, defaultConfig } from '@proj/shared/config';
import { Schema } from 'convict';
import { AppConfig } from './app/app-config';

export default config({
  ...defaultConfig,
  swagger: {
    title: {
      doc: 'Swagger documentation title',
      default: 'NestJS Application',
      format: String,
    },
    description: {
      doc: 'Swagger documentation description',
      default: 'API Documentation',
      format: String,
    },
    version: {
      doc: 'Swagger documentation version',
      default: '0.0.1',
      format: String,
    },
  },
} as Schema<AppConfig>);
"
`;

exports[`swagger generator swagger generator with convict should add Swagger types at app-config 1`] = `
"import { BaseConfig } from '@proj/shared/config';

export interface AppConfig extends BaseConfig {
  swagger: {
    title: string;
    description: string;
    version: string;
  };
}
"
`;
